nixos/modules/services/x11/imwheel.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.imwheel;
+  rcfile = pkgs.writeText "imwheelrc" (concatStringsSep "\n\n"
+    (mapAttrsToList
+      (rule: conf: "\"${rule}\"\n${conf}") cfg.rules
+    ));
+in
+  {
+    options = {
+      services.xserver.imwheel = {
+        enable = mkEnableOption "IMWheel service";
+
+        options = mkOption {
+          type = types.listOf types.str;
+          default = ["-d" "-b 45" "--kill"];
+          example = [ "--sensitivity 1" ];
+          description = ''
+            Additional command-line arguments to pass to
+            <command>imwheel</command>.
+          '';
+        };
+
+        rules = mkOption {
+          type = types.attrsOf types.str;
+          default = {};
+          example = literalExample ''
+            ".*" = \'\'
+              None,      Up,   Button4, 8
+              None,      Down, Button5, 8
+              Shift_L,   Up,   Shift_L|Button4, 4
+              Shift_L,   Down, Shift_L|Button5, 4
+              Control_L, Up,   Control_L|Button4
+              Control_L, Down, Control_L|Button5
+            \'\';
+          '';
+          description = "Configuration";
+        };
+      };
+    };
+
+    config = mkIf cfg.enable {
+      environment.systemPackages = with pkgs; [ imwheel ];
+
+      nixpkgs.overlays = singleton (self: super: {
```

Comment:
Why is this needed? Why does it read the rcfile from within its nix store path and not `/etc/X11/imwheel/imwheelrc`?
I see there's a patch here
```
substituteInPlace Makefile.in --replace "ETCDIR = " "ETCDIR = $out"
substituteInPlace util.c --replace "/etc/X11/imwheel" "$out/etc/X11/imwheel"
```

Which works on `util.c`, we should probably remove this and perhaps use `installFlags` to make it install the file to the nix store.

Reply:
yes that would be an option but it requires change in the definition of the package.

Reply:
Correct, I believe this change is needed for this to function correctly.
All packages that read a config from /etc in nixpkgs are compiled with sysconfdir = /etc and if they have a configuration in the package installed to `$out/etc`. This is because they're configured with a nixos module.

Reply:
sory it took so long @worldofpeace I needed to find time to fiddel with this. Please check if this is what you've ment. No need to hurry though.